lib/imagemaster3000/verification/hash.rb

```diff
@@ -3,42 +3,25 @@ module Verification
     module Hash
       def verify_hash!
         logger.debug 'Verifying checksum'
-        checksum = find_checksum!
         computed_checksum = verification[:hash][:function].file(local_filename).hexdigest
-
-        if checksum == computed_checksum
+        if checksum_present? computed_checksum
           verification[:hash][:checksum] = ::Digest::SHA512.file(local_filename).hexdigest
           return
         end
 
         raise Imagemaster3000::Errors::VerificationError,
-              "Checksum mismatch for file #{local_filename}: expected: #{checksum}, was: #{computed_checksum}"
+              "Not exactly one checksum found for file #{local_filename}: expected: #{checksum}, was: #{computed_checksum}"
```

Comment:
There is no variable `checksum`, it will raise an exception (a different one).
And the message is misleading anyway. You are not looking for a checksum for a specific file, you're just looking for any checksum matching.